### PR TITLE
Add missing cypress tests for initial Distributed Workloads views, fix minor architecture issues, factor out components for sections of the Workload Status tab

### PR DESCRIPTION
**Pull request by mturley**
Part of https://issues.redhat.com/browse/RHOAIENG-3578. Additional unit tests should be added to complete that task.

Also sets up the structure to enable work to begin on https://issues.redhat.com/browse/RHOAIENG-2881 and https://issues.redhat.com/browse/RHOAIENG-3583.

## Description

Followup to https://github.com/opendatahub-io/odh-dashboard/pull/2469. That PR only had tests to ensure the feature flag worked properly. This PR adds further Cypress tests for the global Distributed Workloads page to make sure navigation and project selection work as expected:
* Automatic selection of a project if you visit the `/distributedWorkloads` root
* Clicking tabs to navigate between them and make sure the routes and the page contents update correctly
* Making sure the user-selected project is retained when navigating between tabs and is set as the "preferred project" (it is used if navigating back to the `/distributedWorkloads` root in the same session after choosing a project)
* Making sure the "no projects" empty state appears properly

Running these tests revealed the following issues which are also fixed by this PR:
* I forgot to call `useSyncPreferredProject` in the `DistributedWorkloadsContext`, so when the user selected a project in the dropdown it did not get persisted to state where that selection would be retained when navigating to other pages.
* There was an accessibility error related to the `tabContentId` prop of the `<Tab>` elements in `GlobalDistributedWorkloadsTabs` referencing an element that does not exist on the page (it is used as an `aria-controls` HTML attribute, which must refer to an element that exists). This is because of how I structured the components with only one `<TabContent>` rendered at a time (for the active tab).
  * This a11y error wasn't caught before because none of the previous Cypress tests triggered the `cy.testA11y()` call which is now triggered by calling `globalDistributedWorkloads.visit()`.
  * To resolve this, the TabContent for all tabs must exist at all times, but we still only want to render the contents of the active tab to support future lazy-loading. I refactored the JSX such that the `DistributedWorkloadsToolbar`, `PageSection` and `TabContent` previously rendered by each `tab.ContentComponent` are now lifted out and handled by `GlobalDistributedWorkloadsTabs`, with all but the active tab's `TabContent` set to `hidden`. The `tab.ContentComponent` is now limited to rendering children of `TabContentBody`, and is still only mounted for the active tab.

I also had to add new mock utilities for prometheus query results (`mockPrometheusDWQuery` and `mockPrometheusDWQueryRange`). Currently they just return empty data to get us past the spinner/error states on the page in the Cypress tests, but they will eventually return reasonable default data like `mockPrometheusServing` and `mockPrometheusBias` do.

This PR also moves each of the `GlobalDistributedWorkloads*Tab` components into its own folder (`pages/distributedWorkloads/global/projectMetrics` and `pages/distributedWorkloads/global/workloadStatus`), adds a Grid/Card layout for the workload status tab, and adds new components in the `workloadStatus` folder for each card. This will make it easier to start working in parallel on the separate cards.
* As part of this change, instead of the `GlobalDistributedWorkloadsWorkloadStatusTab` component consuming context and handling error/loading states at that level, each of the new card components consumes context and renders its own errors/spinner.

Screenshot 2024-03-12 at 12 16 15 PM

## How Has This Been Tested?

Wrote new Cypress tests, adjusted behavior to fix failing tests, and re-tested associated behavior manually in the browser.

## Test Impact

N/A, this is a testing PR.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by gitdallas — APPROVED** [review]

**Comment by mturley** [comment]
Marking WIP temporarily while I troubleshoot Cypress failures that are happening in CI but not locally. Going to revert the second commit for a sec to see if it was the rebase on main or some change in that commit.

**Comment by mturley** [comment]
Looks like I needed to specify the GET when intercepting the Projects API call now that we've rebased after https://github.com/opendatahub-io/odh-dashboard/pull/2583. Waiting for CI to verify.

**Comment by mturley** [comment]
CI is green again, good to go!

**Review by gitdallas — APPROVED** [review]

File: frontend/src/__tests__/cypress/cypress/e2e/distributedWorkloads/GlobalDistributedWorkloads.cy.ts
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/distributedWorkloads/GlobalDistributedWorkloads.cy.ts:135]: Does `cy.findDropdownItem` work here?

Comment (mturley) [reply frontend/src/__tests__/cypress/cypress/e2e/distributedWorkloads/GlobalDistributedWorkloads.cy.ts]: Good call, I hadn't noticed the usage of that elsewhere. Pushed a change to duplicate what's done in Pipelines (`selectProjectByName` method which uses `findDropdownItem`).

**Comment by mturley** [comment]
Rebased on main and addressed your comment @christianvogt

**Review by lucferbux — APPROVED** [review]
/lgtm

Did a peer review with Mike, everything fine to unblock the task and continue the work.

**Pull request merged**
